Report prealloc in analytics library

ddn0 opened the pull request:
Push prealloc reporting to analytics library where it can be used by all
callers rather than keeping reporting to only the command-line
interface.

Along the way, make execution timer names more consistent.

l-hoang commented on libquery/include/querying/QueryIR.h:
```diff
@@ -0,0 +1,121 @@
+          const CypherCompiler& _cypher_compiler);
+
+  //! Returns a reference to the query graph.
+  QueryGraph& get_query_graph() { return this->query_graph; }
```
Not sure actually because it might be useful to let the future query planner edit the query graph directly (e.g., adding expression trees to nodes, etc.).

As for why it's not const *at this point in time*, I think it should be able to be a const, but I haven't tried it to let the compiler check for me. There shouldn't be anything that edits the query graph from what I remember.

ddn0 committed:
Report prealloc in analytics library

Push prealloc reporting to analytics library where it can be used by all
callers rather than keeping reporting to only the command-line
interface.

Along the way, make execution timer names more consistent.

arthurp commented on libgalois/include/katana/Statistics.h:
```diff
@@ -434,9 +434,30 @@ ReportStatAvg(
+  ReportPageAllocGuard& operator=(ReportPageAllocGuard&&) = delete;
+
+  void Report() {
+    if (reported_) {
```
Why have a guard like this? And if it's needed why not only call this in the destructor to avoid that need?

ddn0 replied:
in some places, just like `Timer`, the timing regions are in the middle of blocks. I didn't want to move around measured regions in this PR, so `Report` is a way to keep the same measurements without too much change to existing code.
